### DISCUSSION TITLE
[Feature] Undo button after delete

### DISCUSSION THREAD

**Discussion by justin13888** [immich-app/immich#9571] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

At least on the web app, after deleting an asset, there's no obvious way to undo a mistake. A toast with an undo button or Ctrl+Z keyboard shortcut would be great. On the backend, it would also be worth making delete requests to have a delay (e.g. 5-10) seconds before executing to avoid I/O in the case the user quickly changes their mind (albeit these are additional states to worry about)

### Platform

- [X] Server
- [X] Web
- [ ] Mobile

**Comment by mertalev** [comment]
It could probably just be handled in the frontend to set a delay before the delete request, canceling it if the user clicks Undo.

**Reply by justin13888** [reply]
True. I see many apps doing that

**Comment by Thyrfing89** [comment]
I would really like the possibility to change a setting that says: Are you really sure you want to delete this picture

**Comment by vilhalmer** [comment]
This would be nice to have in the duplicate resolution tool as well.

**Comment by xCJPECKOVERx** [comment]
This was added in #18729 and can be closed
